### PR TITLE
Update `Badge` tokens

### PR DESCRIPTION
**Pull request by yurm04**
### WHY are these changes introduced?

Fixes https://github.com/Shopify/polaris-internal/issues/1066

### WHAT is this pull request doing?
* Updated semantic tokens in `polaris-tokens`
* Creating two new `badge` specialty tokens
* Updating `Badge` to use new tokens

| Before | After |
| -- | -- | 
|capture-448f740d|capture-c81386f5|

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#install-dependencies-and-build-workspaces)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

### 🎩 checklist

- [ ] Tested a [snapshot](https://github.com/Shopify/polaris/blob/main/documentation/Releasing.md#-snapshot-releases)
- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

**Comment by lgriffee** [comment]
Adding this comment just for project documentation purposes:

| Before | After |
| -- | -- | 
|capture-448f740d|capture-d1e10430|

File: polaris-tokens/src/themes/base/color.ts [hunk 1]
```diff
@@ -1138,4 +1140,10 @@ export const color: {
   'color-video-thumbnail-play-button-text-on-bg-fill': {
     value: colors.gray[1],
   },
+  'color-badge-new-bg-fill': {
+    value: colors.teal[5],
+  },
+  'color-badge-new-text': {
+    value: colors.teal[14],
+  },
```
Comment (lgriffee) [inline polaris-tokens/src/themes/base/color.ts]: Not blocking but I noticed that this file appears to be in alpha order. If we want to preserve that I think these two new values should be listed after `color-backdrop-bg`.

File: polaris-tokens/src/themes/base/color.ts [hunk 2]
```diff
@@ -116,7 +116,8 @@ export type ColorBackgroundAlias =
   | 'nav-bg'
   | 'radio-button-bg-surface-disabled'
   | 'video-thumbnail-play-button-bg-fill-hover'
-  | 'video-thumbnail-play-button-bg-fill';
+  | 'video-thumbnail-play-button-bg-fill'
+  | 'badge-new-bg-fill';
```
Comment (lgriffee) [inline polaris-tokens/src/themes/base/color.ts]: Not blocking but I noticed that this file appears to be in alpha order. If we want to preserve that I think this new values should be listed after `backdrop-bg`.

File: polaris-tokens/src/themes/base/color.ts [hunk 3]
```diff
@@ -221,7 +222,8 @@ export type ColorTextAlias =
   | 'avatar-text-on-bg-fill'
   | 'avatar-three-text-on-bg-fill'
   | 'avatar-two-text-on-bg-fill'
-  | 'video-thumbnail-play-button-text-on-bg-fill';
+  | 'video-thumbnail-play-button-text-on-bg-fill'
+  | 'badge-new-text';
```
Comment (lgriffee) [inline polaris-tokens/src/themes/base/color.ts]: Not blocking but I noticed that this file appears to be in alpha order. If we want to preserve that I think this new values should be listed after `avatar-two-text-on-bg-fill`.

File: .changeset/light-flowers-pay.md [hunk 4]
```diff
@@ -0,0 +1,6 @@
+---
+'@shopify/polaris': minor
+'@shopify/polaris-tokens': minor
+---
+
+Updated semantic tokens `fill-info-secondary`, `text-info`, `fill-success-secondary`, `fill-caution-secondary`, `fill-critical-secondary`.\nCreated new `Badge` tokens `badge-new-bg-fill`, `badge-new-text` and updated in component styles.
```
Comment (lgriffee) [inline .changeset/light-flowers-pay.md]:
```suggestion
Updated semantic tokens `fill-info-secondary`, `text-info`, `fill-success-secondary`, `fill-caution-secondary`, `fill-critical-secondary`.
Created new `Badge` tokens `badge-new-bg-fill`, `badge-new-text` and updated in component styles.
```

**Review by lgriffee — APPROVED** [review]
✨🎉

**Review by heyjoethomas — DISMISSED** [review]
The default badge should not be teal. [This screen](https://www.figma.com/file/nsEyBD1H13Do6mOlbiJh1h/Uplift-Fast-Follows?type=design&node-id=215-28310&mode=design&t=ZYgWlIs1CPzogkvf-11) should show each badge and value.

**Comment by bernardojoaogarcia** [comment]
It seems like the `default` badge and the `enabled` badge got the same token changes as the `new` badge. We need to create 2 tokens for the `new` badge with the teal color values, and then swap the `fill-transparent-secondary` and the` text-secondary` tokens that are specifically mapped to the `new` badge. No changes are being made to the `default` badge or the `enabled` badge.

**Comment by bernardojoaogarcia** [comment]
1. **Token value changes**

Info badge
`fill-info-secondary` Azure 4 -> Azure 5
`text-info` Azure 14 -> Azure 15

Success badge
`fill-success-secondary` Green 3 -> Green 4

Caution badge
`fill-caution-secondary` Yellow 4 -> Yellow 5

Critical badge
`fill-critical-secondary` Red 6 -> Red 7


2. **Create new token**

New Badge (Specialty tokens)
*create 2 new specialty tokens for the new badge:
`badge-new-bg-fill` = Teal 5
`badge-new-text` = Teal 14


3. **Swap New badge tokens**

`fill-transparent-secondary` -> `badge-new-bg-fill`
`text-secondary` -> `badge-new-text`

**Comment by yurm04** [comment]
Sorry gang, I misinterpreted the original issue. I read everything as "new badge tokens" instead of "New Badge" tokens and updated everything to the new tokens instead of just the `toneNew` class.  This is not the first time i've made that mistake and it probably won't be the last 🤦 Thanks for catching

Updated PR and screenshots in the description.

**Review by alex-page — APPROVED** [review]
Seems legit to me. Just would get @bernardojoaogarcia to double check the design changes. Awesome work @yurm04 🚀

**Review by bernardojoaogarcia — APPROVED** [review]
Thanks @yurm04 ✨🚀

**Review by aaronccasanova — APPROVED** [review]

**Comment by yurm04** [comment]
thanks, @bernardojoaogarcia! could you also review and accept the changes in chromatic? I think that's the last task before we can merge this in

**Comment by bernardojoaogarcia** [comment]
@yurm04 Looking through some of the changes in chromatic I noticed that we are currently using the `new` badge on the Nav! I’m pretty sure we wouldn’t want this type of styling in the nav, so I need to first assess whether 1. we should change the nav badges to `default` badges or 2. revert the `new` badge tokens to the original color and keep them visually the same as the `default` badges. Let's figure out the `new` badge first before proceeding. This flew under the radar because they currently have the exact same visual representation. @yesenia-perezcruz @sarahill 


Screenshot 2023-12-08 at 4 53 59 PM

**Comment by bernardojoaogarcia** [comment]
After chatting with @sarahill, we figured it’s probably a good call to ditch the `new` badge tweaks in this PR and just roll back to the original tokens it used. The `new` badge requires more digging and possibly a new **Gen3** pattern to go with it that holistically covers **New features**. @yurm04

**Comment by bernardojoaogarcia** [comment]
We should just omit steps **2** & **3** of this issue :).

**Comment by yurm04** [comment]
@bernardojoaogarcia I updated with the new changes for your review

**Review by bernardojoaogarcia — APPROVED** [review]
Nice, thank you! @yurm04 ✨ 🙏

**Review by lgriffee — APPROVED** [review]
✨

**Pull request merged**